lib/utils/device_info.dart

```diff
@@ -0,0 +1,28 @@
+import 'dart:io';
+import 'package:android_id/android_id.dart';
+import 'package:device_info_plus/device_info_plus.dart';
+import 'package:simple_logger/simple_logger.dart';
+
+class DeviceInfo {
+  static Future<String?> getSerialNumber() async {
+    DeviceInfoPlugin deviceInfoPlugin = DeviceInfoPlugin();
+
+    if (Platform.isAndroid) {
+      const androidId = AndroidId();
+      String? deviceId = await androidId.getId();
+
+      if (deviceId == null) {
+        SimpleLogger().warning("Device ID not found, cannot connect to MQTT.");
```

Comment:
Device info is not exactly related to MQTT, so this error message sounds a bit incorrect